File: minimum-moves-to-equal-array-elements/minimum-moves-to-equal-array-elements.java
class Solution {
    public int minMoves(int[] nums) {
        if (nums == null || nums.length  <= 1) {
            return 0;
        }
 
        int min = nums[0];
        long sum = 0;
        for (int num : nums) {
            sum += num;
            min = Math.min(min, num);
        }
 
        return (int)(sum - nums.length * min);
    }
}

File: 453-minimum-moves-to-equal-array-elements/minimum-moves-to-equal-array-elements.cpp
class Solution {
public:
    int minMoves(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        long long sum = 0;
        int n = nums.size();
        for(int i = 1; i < int(nums.size()); i++)
        {
            sum += nums[i];
        }

        return sum - (long long)(n - 1) * nums[0];
    }
};